x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/index.test.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { createTestRendererMock, MockedFleetStartServices, TestRenderer } from '../../../mock';
+import { PAGE_ROUTING_PATHS, pagePathGetters, PLUGIN_ID } from '../../../constants';
+import { Route } from 'react-router-dom';
+import { CreatePackagePolicyPage } from './index';
+import React from 'react';
+import { CreatePackagePolicyRouteState } from '../../../types';
+import { act } from 'react-test-renderer';
+
+describe('when on the package policy create page', () => {
+  const createPageUrlPath = pagePathGetters.add_integration_to_policy({ pkgkey: 'nginx-0.3.7' });
+  let testRenderer: TestRenderer;
+  let renderResult: ReturnType<typeof testRenderer.render>;
+  const render = () =>
+    (renderResult = testRenderer.render(
+      <Route path={PAGE_ROUTING_PATHS.add_integration_to_policy}>
+        <CreatePackagePolicyPage />
+      </Route>
+    ));
+
+  beforeEach(() => {
+    testRenderer = createTestRendererMock();
+    mockApiCalls(testRenderer.startServices.http);
+    testRenderer.history.push(createPageUrlPath);
+  });
+
+  describe('and Route state is provided via Fleet HashRouter', () => {
+    let expectedRouteState: CreatePackagePolicyRouteState;
+
+    beforeEach(() => {
+      expectedRouteState = {
```

Comment:
is there a reason this is getting redefined each test run? 

Guessing when you put that as the `history.state`, the object is modified when part of `history`? Is that the case?

Reply:
@pzl I don't think its manipulated by `history`, but it could be mutated. that being said, this is a test and likely it could have been just defined as a static, but I opted to just define it on every test render in order to ensure that every run is "fresh" and not run the risk of the object being mutated between test runs.